docs/devguide/dtypes.rst

```diff
@@ -1,2 +1,77 @@
 Implementing a New Data Type
 ============================
+
+COMPAS data types are classes that are based on :class:`compas.data.Data`.
+
+Data types can be serialised to JSON with
+
+* :func:`compas.json_dump`
+* :func:`compas.json_dumps`
+* :func:`compas.json_dumpz`
+
+and deserialised with the corresponding "load" functions
+
+* :func:`compas.json_load`
+* :func:`compas.json_loads`
+* :func:`compas.json_loadz`
+
+All geometry objects and data structures,
+and also, for example, the visualisation scene,
+are serializable data types.
+
+
+Creating a new data type
+========================
+
+A new data type can be created in a few simple steps.
+
+.. code-block:: python
+
+    class CustomData(Data):
+
+        def __init__(self, a=None, b=None, name=None)
```

Comment:
making all arguments optional is something we were forced to do when we still had the `@data.setter`. As we don't have that anymore, I don't think we need to imply this is necessary.

Reply:
personally, i prefer to have all "constructor" parameters defined as keyword arguments.
in any case, the default behaviour is that the data is expanded as `kwargs` during reconstruction.
so we have to check properly what the different options are...

Reply:
sure, but isn't that related more to how you implement `__data__` rather than what's declared in `__init__`? 
shouldn't `None` default only be assigned to arguments that are not mandatory for creating a valid instance of `CustomData`?